frontend/src/pages/home.js

```diff
@@ -0,0 +1,37 @@
+import React, {useEffect, useState} from 'react'
+import axios from "axios";
+import Table from "../components/table";
+
+import {cloneDeep} from 'lodash'
+
+const Home = () => {
+  const [data, setData] = useState([])
+
+  useEffect(() => {
+    axios.get('/stores').then(response => {
+      setData(response.data)
+    })
+  }, [])
+
+  const handleNameChange = (entry, newName) =>
+    axios.put(`/update_store_name/${entry.id}`, {
+      newName: newName
+    }).then(() => {
+      const entryIndex = data.findIndex((elem) => elem.id === entry.id)
+      const newData = cloneDeep(data)
```

Comment:
is there any particular reason to use cloneDeep here?

Reply:
Quoting react docs:

> Never mutate `this.state` directly, as calling `setState()` afterwards may replace the mutation you made. Treat `this.state` as if it were immutable.

Therefore I needed to make sure the copy that I was making was not a reference copy.